server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -340,157 +391,181 @@ public ClusterMetadataManifest writeIncrementalMetadata(
             }
             previousStateIndexMetadataByName.remove(indexMetadata.getIndex().getName());
         }
+        UploadedMetadataResults uploadedMetadataResults;
+        boolean firstUpload = !previousManifest.hasMetadataAttributesFiles();
```

Comment:
move this firstUpload above and use it when checking for updateCoordinationMetadata in the beginning so that it is not spread across the file.

Reply:
Ack